include/CL/sycl/buffer/detail/buffer_base.hpp

```diff
@@ -129,8 +148,39 @@ struct buffer_base : public std::enable_shared_from_this<buffer_base> {
                               is_write_mode);
   }
 
+#ifdef TRISYCL_OPENCL
+  /// Returns the cl_buffer for a given context.
+  boost::compute::buffer get_cl_buffer(cl::sycl::context context) {
+    return buffer_cache[context];
+  }
+
+
+  /** Create the OpenCL buffer and copy in data from the host if it doesn't
+   already exists for a given context or if the buffer is in write mode */
```

Comment:
Should not be in read mode instead?
But actually it is not discard-write mode, so I am unsure it is correct.